### PR TITLE
feat: Join course QR code

### LINKED ISSUE #349: QR code to join a course
## Current invite modal
image

## The new update
image

The download button generates the image on the right.

I am open to suggestions on how to build this.

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.

### PR DESCRIPTION
**Pull request by codiini**
## What does this PR do?

Fixes #349 

image

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- Go to a course, go to the "people" page, click on the "add button", click on the "download QR code" button and the result is the downloaded qr code.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

**Comment by Chifez** [comment]
@codiini nice job. @rotimi-best i was thinking below the title of the course instead of the course url we should have the name of the academy instead, its nicer and having the link there defeats the purpose of a QR code.

Then also for free tier users we would want to embed our classroomio logo should be at the bottom of the card something like ```powered by classroomio``` or just our logo is fine.

**Comment by rotimi-best** [comment]
@Chifez Great points, thanks.

@codiini Can we make 2 improvements to what you've already done.

1. Instead of the course link, we should have the org name. It is in `$currentOrg.name`
2. Only for free users let's show this kind of chip at the bottom of the image

image

**Comment by codiini** [comment]
@Chifez @rotimi-best , Awesome suggestions. I'll make the changes and push them in a couple of minutes

File: apps/dashboard/src/lib/components/Course/components/People/InvitationModal.svelte [hunk 1]
```diff
@@ -187,21 +224,45 @@
...
 
       <div class="relative">
-        <PrimaryButton
-          disablePadding={true}
-          className="text-sm py-2 px-3"
-          variant={VARIANTS.OUTLINED}
-          onClick={copyLink}
+        <button
+          type="button"
+          on:click={() => copyLink()}
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/Course/components/People/InvitationModal.svelte]: I think we can just do `on:click={copyLink}`

**Review by rotimi-best — APPROVED** [review]
In general looks really good thanks. I will test later

**Review by rotimi-best — APPROVED** [review]
Tested and works pretty good.

Great job @codiini 🎉 

and thanks for the support @tunny17  @Chifez

**Pull request merged**